backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -365,4 +418,362 @@ public FutureExperimentRunDAO(Executor executor, FutureJdbi jdbi, UAC uac) {
         .thenCompose(unused -> artifactHandler.deleteArtifacts(runId, optionalKeys), executor)
         .thenCompose(unused -> updateModifiedTimestamp(runId, now), executor);
   }
+
+  private InternalFuture<UserInfo> getCurrentLoginUserInfo() {
+    return FutureGrpc.ClientRequest(
+        uac.getUACService().getCurrentUser(Empty.newBuilder().build()), executor);
+  }
+
+  public InternalFuture<ExperimentRun> createExperimentRun(final CreateExperimentRun request) {
+    // Validate arguments
+    var futureTask =
+        InternalFuture.runAsync(
+            () -> {
+              String errorMessage = null;
+              if (request.getProjectId().isEmpty() && request.getExperimentId().isEmpty()) {
+                errorMessage =
+                    "Project ID and Experiment ID not found in CreateExperimentRun request";
+              } else if (request.getProjectId().isEmpty()) {
+                errorMessage = "Project ID not found in CreateExperimentRun request";
+              } else if (request.getExperimentId().isEmpty()) {
+                errorMessage = "Experiment ID not found in CreateExperimentRun request";
+              }
+
+              if (errorMessage != null) {
+                throw new InvalidArgumentException(errorMessage);
+              }
+            },
+            executor);
+
+    return futureTask
```

Comment:
This is a massive chain of `thenCompose` with a lot of methods that 1) return what was passed to them and 2) returning a future for things that happen immediately. This is an anti-pattern as it makes the code more complicated than it has to be. This needs to be split into actually meaningful methods/lambdas.

Reply:
Updated this massive code with help of you.